### ISSUE TITLE
Incorrect info (probably a typo) in open function documentation

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#12051]
Migrated from [rt.perl.org#112418](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=112418) (status was 'resolved')

Searchable as RT112418$

**Comment by p5pRT** [comment]
### From jpalecek@web.de

#### Created by jpalecek@web.de



Hello\,

\`perldoc \-f open\` says this about opening a pipe&#8203;:

For example\, use either

&nbsp;                    $child\_pid = open\(FROM\_KID\, "|\-"\)   // die "can't fork&#8203;: $\!";

&nbsp;                or
&nbsp;                    $child\_pid = open\(TO\_KID\,   "|\-"\)   // die "can't fork&#8203;: $\!";

That can't be good \(both directions with the same mode\)\. Obviously\, it should be&#8203;:

&nbsp;                    $child\_pid = open\(FROM\_KID\, "\-|"\)   // die "can't fork&#8203;: $\!";

Also\, the second line is not formatted as code snippet on http&#8203;://perldoc\.perl\.org/functions/open\.html

<details><summary>Perl Info</summary>

```

Flags:
     category=docs
     severity=low

Site configuration information for perl 5.14.2:

Configured by Debian Project at Sat Mar  3 16:40:46 UTC 2012.

Summary of my perl5 (revision 5 version 14 subversion 2) configuration:
    
   Platform:
     osname=linux, osvers=2.6.32-5-686-bigmem, archname=i486-linux-gnu-thread-multi-64int
     uname='linux murphy 2.6.32-5-686-bigmem #1 smp mon jan 16 16:42:05 utc 2012 i686 gnulinux '
     config_args='-Dusethreads -Duselargefiles -Dccflags=-DDEBIAN -D_FORTIFY_SOURCE=2 -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security -Werror=format-security -Dldflags= -Wl,-z,relro -Dlddlflags=-shared -Wl,-z,relro -Dcccdlflags=-fPIC -Darchname=i486-linux-gnu -Dprefix=/usr -Dprivlib=/usr/share/perl/5.14 -Darchlib=/usr/lib/perl/5.14 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.14.2 -Dsitearch=/usr/local/lib/perl/5.14.2 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dsiteman1dir=/usr/local/man/man1 -Dsiteman3dir=/usr/local/man/man3 -Duse64bitint -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Ud_ualarm -Uusesfio -Uusenm -Ui_libutil -DDEBUGGING=-g -Doptimize=-O2 -Duseshrplib -Dlibperl=libperl.so.5.14.2 -des'
     hint=recommended, useposix=true, d_sigaction=define
     useithreads=define, usemultiplicity=define
     useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
     use64bitint=define, use64bitall=undef, uselongdouble=undef
     usemymalloc=n, bincompat5005=undef
   Compiler:
     cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fstack-protector -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
     optimize='-O2 -g',
     cppflags='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fstack-protector -fno-strict-aliasing -pipe -I/usr/local/include'
     ccversion='', gccversion='4.6.2', gccosandvers=''
     intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=12345678
     d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
     ivtype='long long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
     alignbytes=4, prototype=define
   Linker and Libraries:
     ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
     libpth=/usr/local/lib /lib/i386-linux-gnu /lib/../lib /usr/lib/i386-linux-gnu /usr/lib/../lib /lib /usr/lib
     libs=-lgdbm -lgdbm_compat -ldb -ldl -lm -lpthread -lc -lcrypt
     perllibs=-ldl -lm -lpthread -lc -lcrypt
     libc=, so=so, useshrplib=true, libperl=libperl.so.5.14.2
     gnulibc_version='2.13'
   Dynamic Linking:
     dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
     cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib -fstack-protector'

Locally applied patches:
     


@INC for perl 5.14.2:
     /home/jirka/perl5/lib/perl5/i486-linux-gnu-thread-multi-64int
     /home/jirka/perl5/lib/perl5/i486-linux-gnu-thread-multi-64int
     /home/jirka/perl5/lib/perl5
     /etc/perl
     /usr/local/lib/perl/5.14.2
     /usr/local/share/perl/5.14.2
     /usr/lib/perl5
     /usr/share/perl5
     /usr/lib/perl/5.14
     /usr/share/perl/5.14
     /usr/local/lib/site_perl
     .


Environment for perl 5.14.2:
     HOME=/home/jirka
     LANG=cs_CZ
     LANGUAGE=
     LD_LIBRARY_PATH (unset)
     LOGDIR (unset)
     PATH=/home/jirka/perl5/bin:/home/jirka/bin:/home/jirka/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games:/opt/JBuilderX/jdk1.4/bin:/usr/local/MuPAD/share/bin
     PERL5LIB=/home/jirka/perl5/lib/perl5/i486-linux-gnu-thread-multi-64int:/home/jirka/perl5/lib/perl5
     PERL_BADLANG (unset)
     PERL_LOCAL_LIB_ROOT=/home/jirka/perl5
     PERL_MB_OPT=--install_base /home/jirka/perl5
     PERL_MM_OPT=INSTALL_BASE=/home/jirka/perl5
     SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

On Thu Apr 12 12&#8203;:02&#8203;:24 2012\, jpalecek@&#8203;web\.de wrote&#8203;:
> 
> This is a bug report for perl from jpalecek@&#8203;web\.de\,
> generated with the help of perlbug 1\.39 running under perl 5\.14\.2\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> Hello\,
> 
> \`perldoc \-f open\` says this about opening a pipe&#8203;:
> 
> For example\, use either
> 
>                     $child\_pid = open\(FROM\_KID\, "|\-"\)   // die "can't
>    fork&#8203;: $\!";
> 
>                 or
>                     $child\_pid = open\(TO\_KID\,   "|\-"\)   // die "can't
>    fork&#8203;: $\!";
> 
> That can't be good \(both directions with the same mode\)\. Obviously\, it
>    should be&#8203;:
> 
>                     $child\_pid = open\(FROM\_KID\, "\-|"\)   // die "can't
>    fork&#8203;: $\!";
> 
> Also\, the second line is not formatted as code snippet on
>    http&#8203;://perldoc\.perl\.org/functions/open\.html

The first problem appears to be fixed already in 5\.16\.  The second
problem I’ve fixed on the sprout/misc\-post\-5\.16 branch\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed by d18fc9db3ba\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
